pep-0617.rst

```diff
@@ -48,53 +48,58 @@ and the *follow sets* of the grammar:
+  with the token *a*, then the parser can tell that this is an invalid program.
+  But if ``A`` could expand to the empty string (called an ε-production),
+  then the parser would recognise a valid empty ``A``,
+  since the next token *b* is in the *follow set*  of ``A``.
 
 
-The Python grammar does not allow ε-productions so the *follow sets* are not
+The current Python grammar does not contain ε-productions, so the *follow sets* are not
 needed when creating the parse tables. Currently, in CPython, a parser generator
 program reads the grammar and produces a parsing table representing a set of
 deterministic finite automata (DFA) that can be included in a C program, the
-parser, which is a pushdown automaton that uses this data to produce a Concrete
+parser. The parser is a pushdown automaton that uses this data to produce a Concrete
 Syntax Tree (CST) sometimes known directly as a "parse tree". In this process, the
 *first sets* are used indirectly when generating the DFAs.
 
-LL(1) parsers and grammars are usually known for being efficient and simple to
-implement and generate, but the reality is that expressing some constructs
-currently present in the Python language is notably difficult or impossible with
-such a restriction. As LL(1) parsers can only look one token ahead to distinguish
+LL(1) parsers and grammars are usually efficient and simple to implement
+and generate. However, it is not possible, under the LL(1) restriction,
+to express certain common constructs in a way natural to the language
+designer and the reader. This includes some in the Python language.
+
+As LL(1) parsers can only look one token ahead to distinguish
 possibilities, some rules in the grammar may be ambiguous. For instance the rule::
 
     rule: A | B
 
 is ambiguous if the *first sets* of both ``A`` and ``B`` have some elements in
-common. This is because if the parser sees a token in the input
-program that both *A* and *B* can start with it is impossible for it to deduce
-which option to expand as no further token of the program can be examined to
-disambiguate. As will be shown later in this document, the current LL(1)-based
+common. When the parser sees a token in the input
+program that both *A* and *B* can start with, it is impossible for it to deduce
+which option to expand, as no further token of the program can be examined to
+disambiguate.
+The rule may be transformed to equivalent LL(1) rules, but then it may
+be harder for a human reader to grasp its meaning.
+Examples later in this document show that the current LL(1)-based
 grammar suffers a lot from this scenario.
 
-Also, it is relevant to note (as other sections of this document will deal with this
-concept) that a given grammar cannot be LL(1) if it is left recursive. A grammar is
-left-recursive if and only if there exists a nonterminal that can derive to a
+Another broad class of rule precluded by LL(1) is the left-recursive rule.
```

Comment:
```suggestion
Another broad class of rules precluded by LL(1) is left-recursive rules.
```

Reply:
I disagree with the first plural ("class of rule" sounds right to me) but I realise [I'm in the minority](https://english.stackexchange.com/questions/5539/types-of-things-vs-types-of-thing). The rest is equivalent or better.